src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,312 @@
+   * if you don't know the correct answer.
+   * @param {boolean|null} value
+   */
+  setIsFromUserAction(value) {
+    assert(SwgClientEvent.isValidIsFromUserAction_(value),
+        'Invalid isFromUserAction in SwgClientEvent (' + value + ')');
+    this.isFromUserAction_ = value;
+  }
+
+  /**An optional parameter bag that can be included to provide additional
+   * details about some events.
+   * @return {Object}
+   */
+  getAdditionalParameters() {
+    return this.additionalParameters_;
+  }
+
+  /**An optional parameter bag that can be included to provide additional
+   * details about some events.  If you want to add or override a property,
+   * use the get method to get a reference to the existing object.
+   * @param {Object} value
+   */
+  setAdditionalParameters(value) {
+    assert(SwgClientEvent.isValidAdditionalParameters_(value),
+        'Invalid additionalParameters in SwgClientEvent (' + value + ')');
+    this.additionalParameters_ = value;
+  }
+
+  /**Returns true if the object is valid.  Otherwise it throws an error if you
+   * pass true and returns false if you don't.
+   * @param {boolean} createError
+   */
+  isValid(createError) {
+    if (!SwgClientEvent.isValidType_(this.eventType_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid event type in SwgClientEvent (' +
+          this.eventType_ + ')');
+    }
+    if (!SwgClientEvent.isValidOriginator_(this.eventOriginator_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid event originator in SwgClientEvent (' +
+          this.eventOriginator_ + ')');
+    }
+    if (!SwgClientEvent.isValidIsFromUserAction_(this.isFromUserAction_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid isFromUserAction in SwgClientEvent (' +
+          this.isFromUserAction_ + ')');
+    }
+    if (!SwgClientEvent.isValidAdditionalParameters_(
+        this.additionalParameters_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid additional parameters in SwgClientEvent (' +
+          this.additionalParameters_ + ')');
+    }
+    return true;
+  }
+}
+
+export class SwgClientEventManager {
+
+  /**True if the value is a function
+   * @param {*} value
+   * @private
+   */
+  static isFunction_(value) {
+    return typeof value === 'function';
+  }
+
+  constructor() {
+    /** @type {!CallBackArray} @private */
+    this.listeners_ = [];
+
+    /** @type {!CallBackArray} @private */
+    this.filterers_ = [];
+  }
+
+  /**Remove all existing filterers and listeners
+   */
+  clear() {
+    this.listeners_ = [];
+    this.filterers_ = [];
+  }
+
+  /**
+   * Ensures the callback function is notified anytime one of the passed
+   * events occurs unless a filterer returns false.
+   * @param {!SwgEventManagerListener} callback
+   * @public
+   */
+  addListener(callback) {
```

Comment:
I like registerEventListerner better.

Are we no longer providing a list of events of interest? so by default they listen to all events? In that case lets make it specific what the callback is listening to
/**
 * Registers a callback to listen to all events
 * @param {SwgClientEventListener} callback
 */
registerEventListener(callback)

or 
/**
 * Registers a listener for events. If no event(s) is specified, the callback
 * is invoked for all events.
 * @param {(?Array<SwgClientEvent>|?SwgClientEvent)} events
 * @param {!SwgEventListener} callback
 */
registerListener(events, callback)

Reply:
I'm potentially amenable adding event but I intentionally took out "Event" from it because you are already accessing it through .getEventManager().addListener() so it seems like having event in there again is duplicative.

I am amenable to making it "register" instead of add but I see little difference between the meanings and add is shorter.

Reply:
addEventListener would have made more sense if you were going to take some listeners for some events and other listeners for other events. It just seems like we are registering a callback for all events. I prefer "register" if both are okay with you.

Reply:
I don't see any difference between add and register but I don't mind changing it.  For consistency I feel like filterers should have the same naming convention though.